tower-http/src/services/fs/serve_dir.rs

```diff
@@ -914,4 +1029,86 @@ mod tests {
         assert_eq!(res.headers()["content-type"], "text/plain");
         let _ = tokio::fs::remove_file(&tmp_filename).await.unwrap();
     }
+
+    #[tokio::test]
+    async fn read_partial_in_bounds() {
+        let svc = ServeDir::new("..");
+        let bytes_start_incl = 9;
+        let bytes_end_incl = 1023;
+
+        let req = Request::builder()
+            .uri("/README.md")
+            .header(
+                "Range",
+                format!("bytes={}-{}", bytes_start_incl, bytes_end_incl),
+            )
+            .body(Body::empty())
+            .unwrap();
+        let res = svc.oneshot(req).await.unwrap();
+
+        let file_contents = std::fs::read("../README.md").unwrap();
+        assert_eq!(res.status(), StatusCode::PARTIAL_CONTENT);
+        assert_eq!(
+            res.headers()["content-length"],
+            file_contents.len().to_string()
+        );
+        assert!(res.headers()["content-range"]
+            .to_str()
+            .unwrap()
+            .starts_with(&format!(
+                "bytes {}-{}/{}",
+                bytes_start_incl,
+                bytes_end_incl,
+                file_contents.len()
+            )));
+        assert_eq!(res.headers()["content-type"], "text/markdown");
+
+        let body = hyper::body::to_bytes(res.into_body()).await.ok().unwrap();
+        let source = Bytes::from(file_contents[bytes_start_incl..=bytes_end_incl].to_vec());
+        assert_eq!(body, source);
+    }
+
+    #[tokio::test]
+    async fn read_partial_rejects_out_of_bounds_range() {
+        let svc = ServeDir::new("..");
+        let bytes_start_incl = 0;
+        let bytes_end_excl = 9999999;
+        let requested_len = bytes_end_excl - bytes_start_incl;
+
+        let req = Request::builder()
+            .uri("/README.md")
+            .header(
+                "Range",
+                format!("bytes={}-{}", bytes_start_incl, requested_len - 1),
+            )
+            .body(Body::empty())
+            .unwrap();
+        let res = svc.oneshot(req).await.unwrap();
+
+        assert_eq!(res.status(), StatusCode::RANGE_NOT_SATISFIABLE);
```

Comment:
Same for the other test cases but do we want to check that we return the expected content range header here even if its not satisfiable? Since we always seem to set that in the service